README.md

```diff
@@ -1,46 +1,373 @@
+        <!-- Show the file name here -->
+        <p class="flex items-center justify-center text-center w-3/4">
+          {{ file.file ? file.file.previewName : '' }}
+        </p>
+      </div>
+    </template>
+  </SingleFileUpload>
+</div>
+```
+
+<img src="./src/assets/images/single-file-uploading.gif"/>
+
+### Multiple File Upload Management
+
+```js
+<script lang="ts">
+import { MultipleFileUpload } from 'file-upload-frontend'
+
+export default {
+  components: {
+    MultipleFileUpload
+  },
+  data() {
+    return {
+      receivedArray: [],
+      isUploading: false // it's a required thing to pass in component
+    }
+  },
+  methods: {
+    receiveFilesArray(array: []) {
+      this.receivedArray = array // here you will get all the files from preview
+    },
+    uploadingStatus(isUploading: any) {
+      this.isUploading = isUploading
+      // add your uploading logic here and when you successfully uploaded all the files, update the value of isUploading = false
+    }
+  }
+}
+</script>
 ```
 
-### Lint with [ESLint](https://eslint.org/)
+### Description
+
+- In the below html code we provided default design for files management, you can modify this design according to your requirements.
+- The `file` slot containing `file` object with keys `previewType`, `previewUrl`, `previewName`, and `isDragging` for showing preview. like,
 
 ```sh
-npm run lint
+file = file: {
+  previewType: 'image',
+  previewUrl: 'data:image/jpeg;base64,/9j/D1AAAACRsdW1pAAAD...',
+  previewName: 'a152148640581.62d918f12a0b4.jpg',
+  isDragging: false
+}
+```
+
+- previewType: - Type of the preview. like, file is image or video
+- previewUrl: - URL of the file preview
+- previewName: - Preview file name
+- isDragging: - You will get it true when you dragging the file on your design
+
+- `props`
+  - :removeBtn="'remove'" - Provide text for remove button, it will show default `x` if you are not using this props.
+  - :uploadBtn="'Upload'" - Provide text for save or upload files button, it will show default `Upload` if you are not using this props.
+  - :progressBtn="'Uploading...'" - Provide text for button when files under the process, it will show default `Uploading...` if you are not using this props.
+  - :isUploading="isUploading" - This is the `required` props. It will be default `false` and when you got the array of files and successfully uploaded all the files,
+    you need to make it `false` again, as we are making it `true` on button click.
+  - :accept="'image/jpg, image/jpeg, image/png, video/mp4, audio/mp3, application/zip'" - You can add frontend validation for selectiong the required type of file. 
+    By default it will select all the type of files.
+  - Dynamic images according to file types. When you select below described file, it will show your given image. (It's not required, we will show default image)
+    - :pdfImg="'../assets/images/pdf-icon.png'" - For pdf files
+    - :textImg="'../assets/images/text-icon.png'" - For text files
```

Comment:
How can we pass multiple files using these properties? What if I have multiple text files?

Reply:
This is not for real file, as we can't show some content in preview. like for text file musiz file. so, it's for placeholder image where are showing in preview. like,
![image](https://github.com/canopas/file-upload-web-frontend/assets/123158106/9d715244-a996-4280-b9f5-5873dd6a22e0)

Reply:
Then i think we need to rename properties like pdfPreview or something else.
It is currently misleading